packages/rn-tester/js/RNTesterAppShared.js

```diff
@@ -0,0 +1,244 @@
+import {RNTesterThemeContext, themes} from './components/RNTesterTheme';
+import {Header} from './components/RNTesterHeader';
+import {RNTesterEmptyBookmarksState} from './components/RNTesterEmptyBookmarksState';
+
+import type {RNTesterTheme} from './components/RNTesterTheme';
+import type {ExamplesList} from './types/RNTesterTypes';
+
+const APP_STATE_KEY = 'RNTesterAppState.v3';
+
+// RNTester App currently uses Async Storage from react-native for storing navigation state
+// and bookmark items.
+// TODO: Add Native Async Storage Module in RNTester
+LogBox.ignoreLogs([new RegExp('has been extracted from react-native')]);
+
+const DisplayIfVisible = ({isVisible, children}) => (
+  <View style={[styles.container, !isVisible && styles.hidden]}>
+    {children}
+  </View>
+);
+
+type ExampleListsContainerProps = $ReadOnly<{|
+  theme: RNTesterTheme,
+  screen: string,
+  title: string,
+  examplesList: ExamplesList,
+  toggleBookmark: (args: {exampleType: string, key: string}) => mixed,
+  handleExampleCardPress: (args: {exampleType: string, key: string}) => mixed,
+  isVisible: boolean,
+|}>;
+
+const ExampleListsContainer = ({
+  theme,
+  screen,
+  title,
+  examplesList,
+  toggleBookmark,
+  handleExampleCardPress,
+  isVisible,
+}: ExampleListsContainerProps) => {
+  const isBookmarkEmpty = examplesList.bookmarks.length === 0;
+
+  return (
+    <DisplayIfVisible isVisible={isVisible}>
+      <Header title={title} theme={theme} />
+      <DisplayIfVisible isVisible={screen === Screens.COMPONENTS}>
+        <RNTesterExampleList
+          sections={examplesList.components}
+          toggleBookmark={toggleBookmark}
+          handleExampleCardPress={handleExampleCardPress}
+        />
+      </DisplayIfVisible>
+      <DisplayIfVisible isVisible={screen === Screens.APIS}>
+        <RNTesterExampleList
+          sections={examplesList.apis}
+          toggleBookmark={toggleBookmark}
+          handleExampleCardPress={handleExampleCardPress}
+        />
+      </DisplayIfVisible>
+      <DisplayIfVisible isVisible={screen === Screens.BOOKMARKS}>
+        {isBookmarkEmpty ? (
+          <RNTesterEmptyBookmarksState />
+        ) : (
+          <RNTesterExampleList
+            sections={examplesList.bookmarks}
+            toggleBookmark={toggleBookmark}
+            handleExampleCardPress={handleExampleCardPress}
+          />
+        )}
+      </DisplayIfVisible>
+    </DisplayIfVisible>
+  );
+};
+
+const RNTesterApp = (): React.Node => {
+  const [state, dispatch] = useAsyncStorageReducer(
+    RNTesterReducer,
+    initialState,
+    APP_STATE_KEY,
+  );
+
+  React.useEffect(() => {
+    getInitialStateFromAsyncStorage(APP_STATE_KEY).then(
+      initialStateFromStorage => {
+        dispatch({
+          type: RNTesterActionsType.INIT_FROM_STORAGE,
+          data: initialStateFromStorage,
+        });
+      },
+    );
+  }, [dispatch]);
+
+  const {openExample, screen, bookmarks, recentlyUsed} = state;
```

Comment:
Typically would see this at the top of the function with the rest of the state.

Reply:
Done!